packages/apple-pay-integration/src/apple-pay-customer-strategy.ts

```diff
@@ -543,29 +543,39 @@ export default class ApplePayCustomerStrategy implements CustomerStrategy {
     }
 
     private async _getBraintreeDeviceData() {
-        const data = await this._braintreeIntegrationService.getDataCollector();
+        const braintreePaymentMethod = this._paymentIntegrationService
+            .getState()
+            .getPaymentMethod(ApplePayGatewayType.BRAINTREE);
 
-        return data.deviceData;
+        if (braintreePaymentMethod?.clientToken) {
+            const data = await this._braintreeIntegrationService.getDataCollector();
+
+            return data.deviceData;
+        }
     }
 
     private async _initializeBraintreeIntegrationService() {
-        const state = await this._paymentIntegrationService.loadPaymentMethod(
-            ApplePayGatewayType.BRAINTREE,
-        );
+        try {
+            await this._paymentIntegrationService.loadPaymentMethod(ApplePayGatewayType.BRAINTREE);
```

Comment:
This seemed to be called in initialize path, is there a reason why we need to load `braintree` payment method.

Reply:
> https://github.com/bigcommerce/checkout-sdk-js/pull/2469#discussion_r1576177822

Is this the same as here? Since we might be doing the same thing twice? Also maybe add a TODO with a JIRA for plans to fix this i.e. avoid making this call and maybe pass the data as part of apple pay config load.

Reply:
@animesh1987 it's not the same. This is the best way to fix the issue at this point. It's a quick fix. I've already talked to our team about it, and it's going to take some time to fix it on the backend side. When another way will be found, the Braintree initialization process will be removed. At the moment we have some issue with generating `clientToken` on the backend side to pass it to the applepay configuration.

Reply:
That seems ok, can we please add a TODO with JIRA to address the issue above so we have a plan.

Reply:
@animesh1987 added TODO without JIRA. We need to create a ticket for backend developers with description. But they are aware of this problem

Reply:
Please create a Jira ticket for tracking purpose.

Reply:
@animesh1987 @bc-peng i have updated PR with adding JIRA ticket